lnprototest/event.py

```diff
@@ -374,8 +374,11 @@ def action(self, runner: "Runner") -> bool:
 
             err = self.message_match(runner, msg)
             if err:
-                raise EventError(self, "{}: message was {}".format(err, msg.to_str()))
-
+                # Format the error message to be more readable
+                error_msg = f"Expected {self.msgtype}, got {msg.messagetype.name}"
+                if hasattr(msg, "fields"):
+                    error_msg += f": {msg.to_str()}"
```

Comment:
I do not understand why we are checking the `fields` here, can you expand more on this part too?